working/333 - shared memory multithreading/proposal.md

```diff
@@ -842,9 +594,7 @@ class Isolate {
   /// Shared isolate contains a copy of the
```

Comment:
> Is this correct?

That is correct.  

> If so, still sounds very restrictive. 

It depends from which angle you are looking at this: if you are purely looking from the perspective of writing pure-Dart in its single-threaded incarnation - yes, it is very restrictive. But there is a different perspective here: this approach actually _lifts_ restrictions which make it impossible to cover interoperability use-cases described at the start of this document in pure Dart. So despite being restrictive in terms of what Dart programmer could actually write - it actually enables Dart programmer to write express more things.

Reply:
I'm looking at it from the perspective of someone who wants to use the feature, and who wants to know which library features I can't safely use.
I can't tell you. If we change nothing in the libraries I can mention a bunch of things that won't work (`Uri.parse`, `List.toString`, `jsonEncode`, `print` and everything in `dart:async`).

It will take serious refactoring to make parts of the platform libraries useful in a state where global variables throw, and I'm not entirely sure what to refactor it to.

(Could we have a period-of-concept version of such an isolate, something you can spawn using an unpublishable `spawnShared` function that works like `Isolate.spawn`? Just to see how much things break.)

Reply:
👍 To start experimenting with this and putting it behind an API that only works if the experiment is on.